velox/core/PlanNode.h

```diff
@@ -4683,31 +4683,73 @@ class MarkDistinctNode : public PlanNode {
   const RowTypePtr outputType_;
 };
 
-/// Optimized version of a WindowNode for a single row_number function with a
-/// limit over sorted partitions.
-/// The output of this node contains all input columns followed by an optional
+/// Optimized version of a WindowNode for a single row_number, rank or
+/// dense_rank function with a limit over sorted partitions. The output of this
+/// node contains all input columns followed by an optional
 /// 'rowNumberColumnName' BIGINT column.
 class TopNRowNumberNode : public PlanNode {
  public:
+  enum class RankFunction {
```

Comment:
@aditi-pandit The `TopNRowNumberNode ` was previously used to handle the `row_number()` function. We now support the `row_number()`, `rank()`, and `dense_rank()` functions. Therefore, can we rename `TopNRowNumberNode `to `TopNRankLikeNode`? Additionally, should we consider changing `RankFunction `to `RankLikeFunction`?

Reply:
@JkSelf : I do want to change the PlanNode and Operator both to use "TopNRank" in naming. But felt I should do the rename in one go after all the code changes are submit. wdyt ?

Reply:
@aditi-pandit I'm fine with this. Could we add a TODO note for these changes? Thanks.